ultralytics/engine/results.py

```diff
@@ -522,7 +522,13 @@ def plot(
                     .contiguous()
                     / 255
                 )
-            idx = pred_boxes.cls if pred_boxes and color_mode == "class" else reversed(range(len(pred_masks)))
+            idx = (
```

Comment:
@RizwanMunawar Also I think it's better to preserve one `idx` variable. Currently we have to create `idx` here because for segmentation we need to pass all the indexes to `annotator.masks`, but for other tasks like `detect` we are indicating `colors(i)` in a for loop, so we need to add the if-else everywhere for different color_mode.
I think we could declare a `color_idx` first with just one if-else and then we reuse this color_idx everywhere else.
Something like:
```python
 idx = (
    pred_boxes.id
    if pred_boxes.id is not None
    else pred_boxes.cls
    if pred_boxes and color_mode == "class"
    else reversed(range(len(pred_masks)))
)
annotator.masks(pred_masks.data, colors=[colors(x, True) for x in idx], im_gpu=im_gpu)
annotator.box_label(
    box,
    label,
    color=colors(color_idx(i), True),
    rotated=is_obb,
)
```
But maybe we could do this in another PR.

Reply:
got it @Laughing-q, let me open the new PR for this. Will send you a review for this 👍🏻 Thanks 🥇